### PR TITLE
video-interface: expose ended event and also add detection for players that implement the interface

### PR DESCRIPTION
**Pull request by aghassemi**

### CODE REVIEW

File: src/video-interface.js [hunk 1]
```diff
@@ -275,6 +275,17 @@ export const VideoEvents = {
   PAUSE: 'pause',
 
   /**
+   * ended
+   *
+   * Fired when the video ends.
+   *
+   * This event should be fired in addition to `pause` when video ends.
+   *
+   * @event pause
```
Comment (wassgha) [inline src/video-interface.js]: nit `@event pause`

**Comment by aghassemi** [comment]
@wassgha could you PTAL/approve? (hopefully your approval still works, but now `teams` setup is different than before)

File: extensions/amp-video/0.1/test/test-amp-video.js [hunk 2]
```diff
@@ -396,13 +396,18 @@ describe(TAG, () => {
...
               assert.fail('Should not have dispatch unmute message twice');
             });
             v.querySelector('video').dispatchEvent(new Event('volumechange'));
             const successTimeout = timer.promise(10);
             return Promise.race([p, successTimeout]);
+          }).then(() => {
+            const video = v.querySelector('video');
+            video.currentTime = video.duration - 0.1;
+            impl.play();
+            return listenOncePromise(v, VideoEvents.ENDED);
```
Comment (wassgha) [inline extensions/amp-video/0.1/test/test-amp-video.js]: Maybe make sure pause is fired too just for consistency with YouTube tests.

**Pull request merged**